test/adapter/json_api/links_test.rb

```diff
@@ -47,6 +47,15 @@ def test_toplevel_links
             assert_equal(expected, hash[:links])
           end
 
+          def test_nil_toplevel_links
+            hash = ActiveModel::SerializableResource.new(
+              @post,
+              adapter: :json_api,
+              links: nil
+            ).serializable_hash
+            assert_equal(nil, hash[:links])
+          end
```

Comment:
should we test effect with `:json` adapter?  Also, I'm wondering if we should even add a `links` key if the value is null.  Not sure per spec though. not a blocker since this pr has been around for a bit